fs_attachment/models/ir_attachment.py

```diff
@@ -373,14 +373,29 @@ def _set_attachment_data(self, asbytes) -> None:  # pylint: disable=missing-retu
     @api.model
     def _storage_file_read(self, fname: str) -> bytes | None:
         """Read the file from the filesystem storage"""
-        fs, _storage, fname = self._fs_parse_store_fname(fname)
-        try:
-            with fs.open(fname, "rb") as f:
-                return f.read()
-        except IOError:
-            _logger.info(
-                "Error reading %s on storage %s", fname, _storage, exc_info=True
-            )
+        fs, storage_code, fname = self._fs_parse_store_fname(fname)
+
+        read_retry_attempts, read_retry_delay = (
+            self.env["fs.storage"].get_read_retry_config(storage_code)
+            if storage_code
+            else (0, 0.0)
+        )
+        for attempt in range(read_retry_attempts + 1):
+            try:
+                with fs.open(fname, "rb") as f:
+                    return f.read()
+            except FileNotFoundError:
+                _logger.info(
+                    "File not found %s on storage %s (attempt %d)",
+                    fname,
+                    storage_code,
+                    attempt + 1,
+                )
+                time.sleep(read_retry_delay)
```

Comment:
So this means FileNotFoundError will never be raised anymore ?

Reply:
For my customer this happens for example when sending emails from invoices with attachments on it, and because of this error the pdf is empty in the email that is sent to the customer. But if a second email is send with the same attachment it succeeds. 

I suspect the file was just uploaded and S3 did not have the time to list the file in all the nodes. That explains why the attachment is accessible a few moments later. I will do more tests today on real env.